pkg/storageincentives/agent.go

```diff
@@ -370,6 +371,8 @@ func (a *Agent) handleClaim(ctx context.Context, round uint64) (bool, error) {
 
 func (a *Agent) handleSample(ctx context.Context, round uint64) (bool, error) {
 
+	a.state.SetLastSelectedRound(round)
```

Comment:
Because handleSample is invoked in the claim phase, the round is likely to be the current (claim) round and not the upcoming round for which the node is selected and the sample is about to be calculated.   This is what triggered the whole shebang with #3850

Reply:
Placing it at the top of `handleClaim` then will be safer?

Reply:
Nope, both handleSample and handleClaim have the same round number and it's the round that is just ending.  During the claim phase, the NEXT round's neighborhood is selected and the sample for that round will begin to be calculated.

I'm leaving it to the powers that be to decide on the proper fix, but simply SetLastSelectedRound(round+1) or handleSample(ctx,round+1) would work.  It's no worse than the (round - 1) check at line 269. which would probably be busted if handleSample(ctx,round+1) is done.

The comment at line 268 about the sample being from the "previous round" is technically incorrect because the sample is actually for THIS round, the calculation was just started in the previous round's claim phase.

So IMNSHO, the fix should be for handleSample(ctx,round+1) and remove the (round-1) in the check at line 269 since the sample would actually be calculated for the proper (e.g. next) round.